nixos/modules/services/monitoring/rustdesk-server.nix

```diff
@@ -0,0 +1,96 @@
+{ lib, pkgs, config, ... }:
+{
+  options.services.rustdesk-server = with lib; with types; {
+    enable = mkEnableOption "enable RustDesk";
+
+    package = mkOption {
+      type = package;
+      description = "The directus package to use";
+      default = pkgs.rustdesk-server;
+    };
+
+    openFirewall = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Open the connection ports.
+        TCP (21115, 21116, 21117, 21118, 21119)
+        UDP (21116)
+      '';
+    };
+
+    relayIP = mkOption {
+      type = str;
+      description = ''
+        The RustDesk relay has to know its public facing IP
+      '';
+    };
+  };
+
+  config = let
+    cfg = config.services.rustdesk-server;
+    serviceDefaults = {
+      enable = true;
+      requiredBy = [ "rustdesk.target" ];
+      serviceConfig = {
+        Slice = "rustdesk.slice";
+        User  = "rustdesk";
+        Group = "rustdesk";
+        Environment = [];
+        WorkingDirectory = "/var/lib/rustdesk";
+        StateDirectory   = "rustdesk";
+        StateDirectoryMode = "0750";
+        LockPersonality = true;
+        NoNewPrivileges = true;
+        PrivateDevices = true;
+        PrivateMounts = true;
+        PrivateTmp = true;
+        PrivateUsers = true;
+        ProtectClock = true;
+        ProtectControlGroups = true;
+        ProtectHome = true;
+        ProtectHostname = true;
+        ProtectKernelLogs = true;
+        ProtectKernelModules = true;
+        ProtectKernelTunables = true;
+        ProtectProc = "invisible";
+        ProtectSystem = "strict";
+        RemoveIPC = true;
+        RestrictNamespaces = true;
+        RestrictSUIDSGID = true;
+      };
+    };
+  in lib.mkIf cfg.enable {
+    users.users.rustdesk = {
+      description = "System user for RustDesk";
+      isSystemUser = true;
+      group = "rustdesk";
+    };
+    users.groups.rustdesk = {};
```

Comment:
Just to make sure - have you tried making this work with systemds `DynamicUser`? It is usually preferable to creating system users, but isn't always viable due to misc. circumstances.

Reply:
DynamicUser and system user creation can be combined - in that case only the extra isolation of DynamicUser is applied (and if database socket authentication is used then it's usually required aswell to have a fixed system user instead of just DynamicUser)

also you don't really need to get rid of the security flags you already set

Reply:
I'm unfortunately unable to test this right now due to missing hardware. Is someone able to test this patch?
```diff
diff --git a/nixos/modules/services/monitoring/rustdesk-server.nix b/nixos/modules/services/monitoring/rustdesk-server.nix
index a4d85be3caec..885fb608af99 100644
--- a/nixos/modules/services/monitoring/rustdesk-server.nix
+++ b/nixos/modules/services/monitoring/rustdesk-server.nix
@@ -35,15 +35,16 @@ in {
         Slice = "system-rustdesk.slice";
         User  = "rustdesk";
         Group = "rustdesk";
+        DynamicUser = "yes";
         Environment = [];
         WorkingDirectory = "/var/lib/rustdesk";
         StateDirectory   = "rustdesk";
         StateDirectoryMode = "0750";
         LockPersonality = true;
-        NoNewPrivileges = true;
+        # NoNewPrivileges = true; # implied by DynamicUser
         PrivateDevices = true;
         PrivateMounts = true;
-        PrivateTmp = true;
+        # PrivateTmp = true; # implied by DynamicUser
         PrivateUsers = true;
         ProtectClock = true;
         ProtectControlGroups = true;
@@ -53,10 +54,10 @@ in {
         ProtectKernelModules = true;
         ProtectKernelTunables = true;
         ProtectProc = "invisible";
-        ProtectSystem = "strict";
-        RemoveIPC = true;
+        # ProtectSystem = "strict"; # implied by DynamicUser
+        # RemoveIPC = true; # implied by DynamicUser
         RestrictNamespaces = true;
-        RestrictSUIDSGID = true;
+        # RestrictSUIDSGID = true; # implied by DynamicUser
       };
     };
   in lib.mkIf cfg.enable {
```

Reply:
Just tested this and it works.